### PR TITLE
current state of 'Send'- and 'Translate'-module to master

### PR DESCRIPTION
**Pull request by FrederikWolter**

### CODE REVIEW

**Review by FrederikWolter — COMMENTED** [review]
fixed unwanted java version change by SOMEONE.

File: .idea/misc.xml [hunk 1]
```diff
@@ -3,4 +3,4 @@
   <component name="ProjectRootManager" version="2" languageLevel="JDK_17" project-jdk-name="17" project-jdk-type="JavaSDK">
     <output url="file://$PROJECT_DIR$/out" />
   </component>
-</project>
\ No newline at end of file
+</project>
```
Comment (Blo0dR0gue) [inline .idea/misc.xml]: Nothing changed. so okay :D

File: README.md [hunk 2]
```diff
@@ -1,7 +1,8 @@
 # MorseProject
 
-This Project is going to be created in the context of the lecture 'Software Engineering' in the winter semester 2021 in the course Applied Computer Science ('Angewandte Informatik') of the DHBW Mannheim.
-The goal is the transmission of Morse code between two or more devices via sound.
+This Project is going to be created in the context of the lecture 'Software Engineering' in the winter semester 2021 in
+the course Applied Computer Science ('Angewandte Informatik') of the DHBW Mannheim. The goal is the transmission of
+Morse code between two or more devices via sound.
```
Comment (Blo0dR0gue) [inline README.md:5]: Same text only with breaks, but looks okay. We need to add the contributors later anyway.

File: src/com/dhbw/MorseProject/MorseProject.java [hunk 3]
```diff
@@ -1,4 +1,64 @@
...
+import com.dhbw.MorseProject.translate.Translator;
+
+// todo comment ids from Pflichtenheft ;) where these are implemented
+// todo setup jar export
+
+/**
+ * Main entry point to application - providing the main method.
+ * @author Frederik Wolter
+ */
 public class MorseProject {
```
Comment (Blo0dR0gue) [inline src/com/dhbw/MorseProject/MorseProject.java:14]: classe looks okay for the first push.

File: src/com/dhbw/MorseProject/MorseProject.java [hunk 4]
```diff
@@ -1,4 +1,64 @@
 package com.dhbw.MorseProject;
 
+import com.dhbw.MorseProject.send.Encoder;
+import com.dhbw.MorseProject.send.Melody;
+import com.dhbw.MorseProject.translate.Translator;
+
+// todo comment ids from Pflichtenheft ;) where these are implemented
+// todo setup jar export
```
Comment (Blo0dR0gue) [inline src/com/dhbw/MorseProject/MorseProject.java:8]: TODOs in main branch are not common, but it should be okay as this has nothing to do with the code itself.

File: src/com/dhbw/MorseProject/send/Encoder.java [hunk 5]
```diff
@@ -1,4 +1,184 @@
 package com.dhbw.MorseProject.send;
 
+import com.dhbw.MorseProject.translate.Translator;
+
+import javax.sound.sampled.AudioFormat;
+import javax.sound.sampled.AudioSystem;
+import javax.sound.sampled.LineUnavailableException;
+import javax.sound.sampled.SourceDataLine;
+
+// todo comments
```
Comment (Blo0dR0gue) [inline src/com/dhbw/MorseProject/send/Encoder.java:10]: TODOs in main branch are not common, but it should be okay as this has nothing to do with the code itself.

File: src/com/dhbw/MorseProject/send/Encoder.java [hunk 6]
```diff
@@ -1,4 +1,184 @@
...
+
+/**
+ * Class responsible for encoding morse-signals to audio and therefor generating and sending the audio signal.
+ * @author Lucas Schaffer & Frederik Wolter
+ */
 public class Encoder {
```
Comment (Blo0dR0gue) [inline src/com/dhbw/MorseProject/send/Encoder.java:16]: Looks okay.

File: src/com/dhbw/MorseProject/translate/Translator.java [hunk 7]
```diff
@@ -4,18 +4,32 @@
 
 
 /**
- * @author Hassan El-Khalil
+ * @author Hassan El-Khalil, (Frederik Wolter)
```
Comment (Blo0dR0gue) [inline src/com/dhbw/MorseProject/translate/Translator.java]:
```suggestion
 * @author Hassan El-Khalil, supported by Frederik Wolter
```

File: src/com/dhbw/MorseProject/send/Melody.java [hunk 8]
```diff
@@ -0,0 +1,57 @@
+package com.dhbw.MorseProject.send;
+
+import java.util.ArrayList;
+
+/**
+ * This class is used as a POJO for the representation of a melody object.
+ * @author Frederik Wolter
+ */
+public class Melody {
```
Comment (Blo0dR0gue) [inline src/com/dhbw/MorseProject/send/Melody.java:9]: class looks good. Love this regions :D

File: src/com/dhbw/MorseProject/translate/Translator.java [hunk 9]
```diff
@@ -58,69 +72,73 @@
...
         CharToMorse.put(' ', W);
-        reverse();
+        // endregion
+
+        reverse();  // initialize MorseToChar by reversing CharToMorse
     }
 
     /**
-     *This Method reverts the CharToMorse Hashmap
+     * This Method reverts the CharToMorse Hashmap and saves it in MorseToChar.
```
Comment (Blo0dR0gue) [inline src/com/dhbw/MorseProject/translate/Translator.java]:
```suggestion
     * This Method reverse the CharToMorse Hashmap and saves it in MorseToChar.
```

**Review by Blo0dR0gue — APPROVED** [review]
Looks okay in general. Just some small suggestions. 
I also can't delete the empty file "tbd" that @ghghrzg pushed. Would be nice if someone else could remove this file before we merge it. I will accept the request anyway :D

File: src/com/dhbw/MorseProject/MorseProject.java (hunk 4 above)
Comment (FrederikWolter) [reply src/com/dhbw/MorseProject/MorseProject.java:8]: better than no code ;)

**Review by kaigru — APPROVED** [review]

**Pull request merged**